src/frontend/contexts/LocationStoreContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {createContext, useContext, useState} from 'react';
+import {
+  Accuracy,
+  LocationObject,
+  LocationSubscription,
+  useForegroundPermissions,
+  watchPositionAsync,
+} from 'expo-location';
+
+const LocationStoreContext = createContext<LocationStore | null>(null);
+
+// TODO: Return undefined when provider status changes to unavailable
+export function LocationStoreProvider({children}: {children: React.ReactNode}) {
+  const [store] = useState(() => new LocationStore());
+
+  const [locationPermissions] = useForegroundPermissions();
+
+  if (locationPermissions?.granted) {
+    store.init();
+  }
+
+  return (
+    <LocationStoreContext.Provider value={store}>
+      {children}
+    </LocationStoreContext.Provider>
+  );
+}
+
+export function useLocationStore() {
+  const store = useContext(LocationStoreContext);
+
+  if (!store) {
+    throw new Error(
+      'No LocationStore set, use LocationStoreProvider to set one',
+    );
+  }
+
+  return store;
+}
+
+export interface LocationState {
+  location: LocationObject | null;
+  error: Error | null;
+}
+
+export class LocationStore {
+  #locationSubscription: LocationSubscription | null = null;
+  #subscribers = new Set<() => void>();
+  #state: LocationState = {
+    location: null,
+    error: null,
+  };
+
+  /**
+   * Initialize position watching. Permissions handling should be done before calling this.
+   */
+  async init() {
+    try {
+      if (this.#locationSubscription) return;
+      this.#locationSubscription = await watchPositionAsync(
+        {accuracy: Accuracy.BestForNavigation},
```

Comment:
based on the call we had, should update this so that it fires only once per second at most

```suggestion
        {accuracy: Accuracy.BestForNavigation, timeInterval: 1000},
```